TASK-68: Searchable Events Frontend

JimLau49 opened the pull request:
This PR closes #232 

Prereq: You should have upcoming events and hosted/cohosted events before testing this

To test:

1. `nx serve mull-ui` and `nx serve mull-api`
2. Go to localhost:4200/event-messages-list
3. Type in the search bar to filter the events 

![image](https://user-images.githubusercontent.com/35787279/109426838-9a556680-79bd-11eb-9590-cf43fcef411a.png)

Notes:
Events are currently a static array, will be changed to a query for event channels in a future ticket.

JimLau49 committed:
feat: event bullet custom component

refs #232

JimLau49 committed:
feat: searchable event message list page

refs #232

bit172 commented on apps/mull-ui/src/app/components/event-bullet/event-bullet.scss:
```diff
@@ -0,0 +1,35 @@
+@import '../../../variables.scss';
```
Who's Rem?

bit172 commented on apps/mull-ui/src/app/components/event-bullet/event-bullet.tsx:
```diff
@@ -0,0 +1,46 @@
+  eventDate: Date;
+}
+
+const monthNames = [
+  'Jan',
+  'Feb',
+  'March',
+  'April',
+  'May',
+  'June',
+  'July',
+  'Aug',
+  'Sept',
+  'Oct',
+  'Nov',
+  'Dec',
+];
```
Move this to constants

bit172 commented on apps/mull-ui/src/app/components/event-bullet/event-bullet.tsx:
```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import './event-bullet.scss';
+
+/* eslint-disable-next-line */
```
This can be removed

bit172 commented on apps/mull-ui/src/app/components/event-bullet/event-bullet.tsx:
```diff
@@ -0,0 +1,46 @@
+    <div>
+      <div className="event-bullet-container">
+        <img className="event-photo" src={eventPicture} alt="event"></img>
+
```
Why is there whitespace here?

bit172 commented on apps/mull-ui/src/app/components/event-bullet/event-bullet.tsx:
```diff
@@ -0,0 +1,46 @@
+        <img className="event-photo" src={eventPicture} alt="event"></img>
+
+        <div className="event-bullet-date-box">
+          <div className="event-box-day-font">{eventDate.getDate()}</div>
+          <div className="event-box-font">{monthNames[eventDate.getMonth()]}</div>
```
Consider using p/span tags here instead of div's

bit172 commented on apps/mull-ui/src/app/pages/messages/event-messages/event-message-list.tsx:
```diff
@@ -0,0 +1,60 @@
+        {messageChannels.map((messageChannel) => {
+          if (messageChannel.eventTitle.toLowerCase().includes(searchString.toLowerCase()))
+            return (
+              <EventBullet
+                eventTitle={messageChannel.eventTitle}
+                eventPicture={messageChannel.eventPicture}
+                eventDate={messageChannel.eventDate}
+              />
```
Add a key prop here

bit172 commented on apps/mull-ui/src/app/pages/messages/event-messages/event-message-list.tsx:
```diff
@@ -0,0 +1,60 @@
+  return (
+    <div className="event-messages-container">
+      <ChatHeader eventTitle="Event Messages" />
+      <div className="page-container">
```
This div should be the first one

JimLau49 replied:
This is what i did to switch the divs around to retain styling, lmk if theres a better way to do it since the Chat-Header had a style prop i added some styling there
![image](https://user-images.githubusercontent.com/35787279/109441833-5cc7fc00-7a04-11eb-9677-d5e2b3ca972d.png)

bit172 commented on apps/mull-ui/src/app/pages/messages/event-messages/event-message-list.tsx:
```diff
@@ -0,0 +1,60 @@
+          placeholder="Search"
+        />
+        {messageChannels.map((messageChannel) => {
+          if (messageChannel.eventTitle.toLowerCase().includes(searchString.toLowerCase()))
```
Do you need to have a case for the empty string? Check @leojrsilao's file [Here](https://github.com/RGPosadas/Mull/blob/43a1593990209fee64f56b4b6593735cc03e9d4d/apps/mull-ui/src/app/pages/direct-message/direct-message.tsx)

bit172 replied:
Also, we could probably add some no results text if it doesn't find anything.

JimLau49 replied:
Empty string will just display all the events channels that it has available (static for now)

bit172 reviewed (changes requested):
Good work! Just a few small issues I found.

JimLau49 committed:
fix: address pr comments

refs #232

osahtout commented on apps/mull-ui/src/app/pages/messages/event-messages/event-message-list.tsx:
```diff
@@ -0,0 +1,64 @@
+
+  return (
+    <div className="page-container">
+      <ChatHeader eventTitle="Event Messages" style={{ right: '0.625rem', marginTop: '1rem' }} />
```
Cindy had a great suggestion for the header in Leo's PR. https://github.com/RGPosadas/Mull/pull/243/files#r584393398. This should fix the gap between the top nav and the header
![Image from iOS](https://user-images.githubusercontent.com/44760251/109523997-f8e21980-7a7d-11eb-8b60-5b930d2305d6.png)

JimLau49 replied:
Ya this is a good suggestion, im gonna do this

osahtout commented on apps/mull-ui/src/app/pages/messages/event-messages/event-message-list.tsx:
```diff
@@ -0,0 +1,64 @@
+  },
+  {
+    eventTitle: 'Lets do this guys!',
+    eventPicture: 'https://i.pinimg.com/originals/c9/a1/44/c9a144b70e175c0361c6e2725677e55a.jpg',
```
is it possible to have one landscape, one portrait and one square photo to see the effect?

JimLau49 replied:
sure thing ill look for one

osahtout commented on apps/mull-ui/src/app/components/event-bullet/event-bullet.scss:
```diff
@@ -0,0 +1,33 @@
+}
+
+.event-photo {
+  max-width: 3rem;
+  max-height: 4rem;
```
I'm not sure if having different height and width is a good idea. can it cause trouble for portrait of square photos.

JimLau49 replied:
changing both to 3rem

osahtout commented on apps/mull-ui/src/app/components/event-bullet/event-bullet.scss:
```diff
@@ -0,0 +1,33 @@
+}
+
+.event-photo {
+  max-width: 3rem;
```
is it possible to match the height of the date-box? having 4rem does seems to work

JimLau49 replied:
![image](https://user-images.githubusercontent.com/35787279/109536564-e078fb80-7a8b-11eb-9500-659901907f57.png)
Height adjusted

JimLau49 committed:
feat: add chat header to top-nav + adjust iconsize

refs #232

bit172 commented on apps/mull-ui/src/app/components/event-bullet/event-bullet.scss:
```diff
@@ -0,0 +1,33 @@
+  border-radius: 15%;
+}
+
+.event-bullet-date-box {
+  display: flex;
+  border: 2px #a9a9a9;
+  border-radius: 15%;
+  border-style: solid;
+  flex-direction: column;
+  align-items: center;
+  padding: 2px 10px;
+  margin: 0 10px;
+}
+
+.event-box-day-font {
+  font-size: 12px;
+  font-weight: bold;
+}
+
+.event-box-font {
+  font-size: 10px;
+}
```
These are still in pixels

JimLau49 replied:
My bad forgot to change these

bit172 commented on apps/mull-ui/src/app/components/event-bullet/event-bullet.tsx:
```diff
@@ -0,0 +1,31 @@
+      <div className="event-bullet-container">
+        <img className="event-photo" src={eventPicture} alt="event"></img>
+
+        <div className="event-bullet-date-box">
```
```suggestion
        <img className="event-photo" src={eventPicture} alt="event" />
        <div className="event-bullet-date-box">
```

osahtout commented on apps/mull-ui/src/app/pages/messages/event-messages/event-message-list.tsx:
```diff
@@ -0,0 +1,68 @@
+
+/*TODO: Remove this and replace with query of event message channels*/
+
+const messageChannels = [
```
is it not better to use seeders instead here?

JimLau49 replied:
This is going to be replaced with a query for event channels just like for #243

osahtout replied:
I mean are we not going to query all the events the user is participating in? can't we do that right now?

JimLau49 committed:
chore: change px to rem, remove chat header

refs #232

JimLau49 committed:
chore: change event bullet class names

refs #232

JimLau49 committed:
chore: add todo for swipeable routes

refs #232

JimLau49 commented:
Hey, so going with what was discussed in the meeting last night, i've removed the chat header from the component as it will be replaced with swipeable routes in #225 
![image](https://user-images.githubusercontent.com/35787279/109708227-1dff8680-7b69-11eb-97f3-221a9a569388.png)

bit172 reviewed (approved):
LGTM! 🚀 Should probably add the actual events since the queries already exist.

- [x] Documentation present where necessary
- [x] Unit and UI tests are implemented for every new component and function
- [x] The application should still work with seeded data
- [x] Must match mockups:
  - [Events only](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/Mull?node-id=2087%3A74365)

JimLau49 committed:
feat: add event query for event message list

refs #232
Co-authored-by: Ritchelle Posadas<Ritchelle.Posadas@gmail.com>

JimLau49 committed:
fix: remove use of fragments in queries

This was done because MockedProvider was able to get the passed data
from fragments  -->
failing the new snapshot tests

refs #232

Co-authored-by: Ritchelle Posadas<Ritchelle.Posadas@gmail.com>

JimLau49 committed:
test: update event-message-list tests

refs #232

Co-authored-by: Ritchelle Posadas<Ritchelle.Posadas@gmail.com>

JimLau49 committed:
chore: fix linting errors

refs #232

Co-authored-by: Ritchelle Posadas<Ritchelle.Posadas@gmail.com>

JimLau49 committed:
chore: fix linting errors

refs #232

Co-authored-by: Ritchelle Posadas<Ritchelle.Posadas@gmail.com>

JimLau49 committed:
Merge branch 'TASK-68/searchable-events-frontend' of https://github.com/RGPosadas/Mull into TASK-68/searchable-events-frontend

RGPosadas reviewed (approved):
LGTM! 🚀 

**Acceptance Criteria**
- [x] Documentation present where necessary
- [x] Unit and UI tests are implemented for every new component and function
- [x] The application should still work with seeded data
- [x] Must match mockups:
  - [Events only](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/Mull?node-id=2087%3A74365)
